jaxley/modules/base.py

```diff
@@ -1697,578 +2046,314 @@ def _rotate(self, degrees: float, rotation_axis: str, view: pd.DataFrame):
-    ):
-        """Append multiple rows to the `self.edges` table.
-
-        This is used, e.g. by `fully_connect` and `connect`.
-
-        Args:
-            pre_rows: The pre-synaptic compartments.
-            post_rows: The post-synaptic compartments.
-            synapse_type: The synapse to append.
-
-        both `pre_rows` and `post_rows` can be obtained from self.view.
-        """
-        # Add synapse types to the module and infer their unique identifier.
-        synapse_name = synapse_type._name
-        index = len(self.pointer.edges)
-        type_ind, is_new = self._infer_synapse_type_ind(synapse_name)
-        if is_new:  # synapse is not known
-            self._update_synapse_state_names(synapse_type)
-
-        post_loc = loc_of_index(
-            post_rows["global_comp_index"].to_numpy(),
-            post_rows["global_branch_index"].to_numpy(),
-            self.pointer.nseg_per_branch,
-        )
-        pre_loc = loc_of_index(
-            pre_rows["global_comp_index"].to_numpy(),
-            pre_rows["global_branch_index"].to_numpy(),
-            self.pointer.nseg_per_branch,
-        )
-
-        # Define new synapses. Each row is one synapse.
-        new_rows = dict(
-            pre_locs=pre_loc,
-            post_locs=post_loc,
-            pre_branch_index=pre_rows["branch_index"].to_numpy(),
-            post_branch_index=post_rows["branch_index"].to_numpy(),
-            pre_cell_index=pre_rows["cell_index"].to_numpy(),
-            post_cell_index=post_rows["cell_index"].to_numpy(),
-            type=synapse_name,
-            type_ind=type_ind,
-            global_pre_comp_index=pre_rows["global_comp_index"].to_numpy(),
-            global_post_comp_index=post_rows["global_comp_index"].to_numpy(),
-            global_pre_branch_index=pre_rows["global_branch_index"].to_numpy(),
-            global_post_branch_index=post_rows["global_branch_index"].to_numpy(),
-        )
-
-        # Update edges.
-        self.pointer.edges = concat_and_ignore_empty(
-            [self.pointer.edges, pd.DataFrame(new_rows)],
-            ignore_index=True,
+        elif not has_node_inds and has_edge_inds:
+            base_nodes = self.base.nodes
+            self._edges_in_view = at_edges
+            incl_comps = pointer.edges.loc[
+                self._edges_in_view, ["global_pre_comp_index", "global_post_comp_index"]
+            ]
+            incl_comps = np.unique(incl_comps.to_numpy().flatten())
+            where_comps = base_nodes["global_comp_index"].isin(incl_comps)
+            possible_nodes_in_view = base_nodes.index[where_comps].to_numpy()
+            self._nodes_in_view = np.intersect1d(
+                possible_nodes_in_view, self._nodes_in_view
+            )
+        elif has_node_inds and has_edge_inds:
+            self._nodes_in_view = at_nodes
+            self._edges_in_view = at_edges
+
+    # TODO: Implement this
```

Comment:
remove the code for now? Or at least add a docstring what this should do and why it would be good to have.